custom-notification-handler/src/main/java/org/wso2/carbon/sample/internal/CustomNotificationHandlerServiceComponent.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.sample.internal;
+
+import org.wso2.carbon.sample.CustomNotificationHandler;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.wso2.carbon.identity.event.handler.AbstractEventHandler;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.user.core.service.RealmService;
+
+/**
+ * @scr.component name="identity.event.custom.handler.notification" immediate="true
```

Comment:
Can we refactor the OSGI component structure as implemented here[1]?.

[1] https://github.com/wso2-extensions/identity-governance/blob/e8acf50de02080bcd954376a6c1430d4ba03b0a4/components/org.wso2.carbon.identity.user.export.core/src/main/java/org/wso2/carbon/identity/user/export/core/internal/service/impl/BasicUserInformationProvider.java#L50